app/src/main/java/com/android/streetworkapp/MainActivity.kt

```diff
@@ -269,22 +268,15 @@ fun StreetWorkApp(
   Scaffold(
       containerColor = ColorPalette.PRINCIPLE_BACKGROUND_COLOR,
       topBar = {
-        if (showSearchBar.value && screenParams?.hasSearchBar == true) {
-          MapSearchBar(searchQuery) {
-            searchQuery.value = ""
-            showSearchBar.value = false
-          }
-        } else {
-          screenParams
-              ?.isTopBarVisible
-              ?.takeIf { it }
-              ?.let {
-                TopAppBarWrapper(navigationActions, screenParams?.topAppBarManager)
-                // setup the InfoDialogManager in topBar, because it relies on the topAppBarManager.
-                Log.d("InfoDialog", "Main - Setting up the InfoDialogManager")
-                infoManager.setUp()
-              }
-        }
+        screenParams
+            ?.isTopBarVisible
+            ?.takeIf { it }
+            ?.let {
+              TopAppBarWrapper(navigationActions, screenParams?.topAppBarManager, searchQuery)
+              // setup the InfoDialogManager in topBar, because it relies on the topAppBarManager.
+              Log.d("InfoDialog", "Main - Setting up the InfoDialogManager")
```

Comment:
Suggestion: I know that it was already there, but for consistency, you could use the tag `MainActivity` instead of `InfoDialog`. The tag could be set as a constant at the beginning of the file.